File: Lab Sheet 5/Rust/src/qD.rs
use text_io::read;
use std::io::BufRead;

fn read_vec<T>() -> Vec<T>
where
    T: std::str::FromStr,
    <T as std::str::FromStr>::Err: std::fmt::Debug,
{
    std::io::stdin()
        .lock()
        .lines()
        .next()
        .unwrap()
        .unwrap()
        .trim()
        .split_whitespace()
        .map(|s| s.parse::<T>().unwrap())
        .collect::<Vec<T>>()
}

fn main() {
    let mut count: usize = 0;
    let mut n: usize;
    let mut k: usize;
    let mut t: usize;
    n = read!();
    k = read!();
    t = read!();

    let mut arr = read_vec::<usize>();

    let mut count: usize;
    let mut sum: usize;    

    sum = 0;
    count = 0;

    //// First window
    for i in 0..k {
        sum += arr[i];
    }

    if sum/k >= t {
        count += 1;
    }

    //// Other windows
    for i in 1..n-k+1 {
        sum += arr[i+k-1];
        sum -= arr[i-1];
        if sum/k >= t {
            count += 1;
        }
    }

    println!("{:?}", count);
}

File: Lab Sheet 6/C/qD.c
#include <stdio.h>

int main()
{
    //// Input

    int n;
    scanf("%d", &n);
    int arr[n];
    int count[n];
    for (int i = 0; i < n; i++)
    {
        scanf("%d", &arr[i]);
        count[i] = 0;
    }

    int j = 0;
    for (int i = 0; i < n; i++)
    {
        count[arr[i]] += 1;
        while (j <= i && count[arr[j]] != 1)
        {
            j++;
        }
        if (j > i)
        {
            printf("-1 ");
        }
        else
        {
            printf("%d ", arr[j]);
        }

        printf("\ni: %d, j: %d\n", i, j);
    }

    return 0;
}